src/charm.py

```diff
@@ -82,6 +72,7 @@ def _on_collect_status(self, event: CollectStatusEvent):  # noqa C901
             return
         try:
             self._charm_config: CharmConfig = CharmConfig.from_charm(charm=self)
+            self._network = self._get_network_configuration()
```

Comment:
Does this belong in the try/except? I.e. could the `CharmConfigInvalidError` come from calling it? If not, let's move it out